pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/SegmentStatusChecker.java

```diff
@@ -175,6 +175,9 @@ public void runSegmentMetrics() {
       }
       IdealState idealState = helixAdmin.getResourceIdealState(helixClusterName, tableName);
       if (idealState == null) {
```

Comment:
Change this condition to cover all cases.

idealState == null || idealState.getPartitionSet().isEmpty()